docs/DeveloperGuide.md

```diff
@@ -439,87 +439,748 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+5. TaskHub shows a success message to the User.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. User enters an invalid command.
+  * 2a1. TaskHub shows an error message.
+
+    Use case ends.
+
+* 2b. User does not enter a name.
+  * 2b1. TaskHub shows an error message.
+    
+    Use case ends.
+
+* 2c. User enters a name that does not exist.
+  * 2c1. TaskHub shows a blank list of employees and projects.
+        
+    Use case ends.
+
+**Use case: UC06 - Add a project**
+
+**MSS**
+
+1. User requests to add a project.
+2. TaskHub adds the project to the list of projects and displays the updated list.
+3. TaskHub shows that the project has been added successfully.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. User enters an invalid command.
+  * 1a1. TaskHub shows an error message.
+    
+    Use case ends.
+
+* 1b. User does not enter a required field.
+  * 1b1. TaskHub shows an error message.
+
+    Use case ends.
+
+* 1c. User enters details of a project that already exists.
+  * 1c1. TaskHub shows an error message.
+
+    Use case ends.
+
+* 1d. User enters an invalid value for a field.
+  * 1d1. TaskHub shows an error message.
+
+    Use case ends.
+
+* 1e. User enters multiple values for a single-valued field.
+  * 1e1. TaskHub shows an error message.
+    
+    Use case ends.
+
+**Use case: UC07 - List all projects**
+
+**MSS**
 
-  Use case ends.
+1. User requests to list all projects.
+2. TaskHub displays the list of projects.
+3. TaskHub shows that all projects have been listed successfully.
 
-* 3a. The given index is invalid.
+    Use case ends.
 
-    * 3a1. TaskHub shows an error message.
+**Extensions**
 
-      Use case resumes at step 2.
+* 1a. User enters an invalid command.
+  * 1a1. TaskHub shows an error message.
 
+    Use case ends.
 
-**Use case 2: Delete a project**
+**Use case: UC08 - Edit a project**
 
 **MSS**
 
-1.  User requests to list projects
-2.  TaskHub shows a list of projects
-3.  User requests to delete a specific project in the list
-4.  TaskHub deletes the employee
+1. User lists all projects (UC07).
+2. User requests to edit the details of a project.
+3. TaskHub edits the project with the new details and displays the updated list.
+4. TaskHub shows that the project has been edited successfully.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 2a. User enters an invalid command.
+  * 2a1. TaskHub shows an error message.
 
-  Use case ends.
+    Use case ends.
 
-* 3a. The given index is invalid.
+* 2b. User does not enter at least one field to edit.
+  * 2b1. TaskHub shows an error message.
+    
+    Use case ends.
 
-    * 3a1. TaskHub shows an error message.
+* 2c. User enters an invalid value for a field.
+  * 2c1. TaskHub shows an error message.
 
-      Use case resumes at step 2.
+    Use case ends.
 
-**Use case 3: Add an employee**
+* 3d. User edits the project's name to one that already exists.
```

Comment:
2d

Reply:
You are correct, it is supposed to be 2d.